app/services/order-service.js

```diff
@@ -1,7 +1,16 @@
 import ApiBaseService from "./api-base-service";
 
 export default ApiBaseService.extend({
+  store: Ember.inject.service(),
+
   ordersCountFor(userId) {
     return this.GET(`/users/${userId}/orders_count`);
+  },
+
+  changeOrderState(order, params) {
+    return this.PUT(`/orders/${order.id}/transition`, params).then(data => {
+      data["designation"] = data["order"];
+      this.get("store").pushPayload(data);
+    });
```

Comment:
Suggestion :

```javascript
changeOrderState(order, state, opts = {}) {
    return this.PUT(`/orders/${order.id}/transition`, { transition: state, ...opts }).then(data => {
      data["designation"] = data["order"]; // todo: consolidate the naming under 'order'
      this.get("store").pushPayload(data);
    });
},

// shorthand method to cancel orders
cancelOrder(order, reason) {
   return this.changeOrderState(order, 'cancel', { reason });
}
```

That allows you to call it like this : 

```
service.changeOrderState(order, 'cancel');
service.cancelOrder(order, 'pure evil');
```

Which is a bit more readable (imo).
This is not a change request, more for the sake of conversation :)